src/ucp/wireup/select.c

```diff
@@ -308,8 +316,8 @@ static UCS_F_NOINLINE ucs_status_t ucp_wireup_select_transport(
         const ucp_wireup_select_params_t *select_params,
         const ucp_wireup_criteria_t *criteria, ucp_tl_bitmap_t tl_bitmap,
         uint64_t remote_md_map, uint64_t local_dev_bitmap,
-        uint64_t remote_dev_bitmap, int show_error,
-        ucp_wireup_select_info_t *select_info)
+        uint64_t remote_dev_bitmap, int show_error, unsigned *local_dev_counts,
+        unsigned *remote_dev_counts, ucp_wireup_select_info_t *select_info)
```

Comment:
this function is already complicated. i think device counts logic should be outside of it.